MatrixSDK/Aggregations/LocationSharing/MXBeaconAggregations.swift

```diff
@@ -0,0 +1,170 @@
+        }
+        
+        guard let beacon = MXBeacon(mxEvent: event) else {
+            return
+        }
+                
+        guard let beaconInfoSummary = self.getBeaconInfoSummary(withIdentifier: beacon.beaconInfoEventId, inRoomWithId: roomId), self.canAddBeacon(beacon, to: beaconInfoSummary) else {
+            return
+        }
+        
+        if beaconInfoSummary.updateWithLastBeacon(beacon) {
+            self.notifyBeaconInfoSummaryListeners(ofRoomWithId: roomId, beaconInfoSummary: beaconInfoSummary)
+        }
+    }
+    
+    public func handleBeaconInfo(event: MXEvent) {
+        guard let roomId = event.roomId else {
+            return
+        }
+        
+        guard let beaconInfo = MXBeaconInfo(mxEvent: event) else {
+            return
+        }
+        
+        self.addOrUpdateBeaconInfo(beaconInfo, inRoomWithId: roomId)
+    }
+    
+    // MARK: Data update listener
+    
+    public func listenToBeaconInfoSummaryUpdateInRoom(withId roomId: String, handler: @escaping (MXBeaconInfoSummaryProtocol) -> Void) -> Any? {
+        let listener = MXBeaconInfoSummaryListener(roomId: roomId, notificationHandler: handler)
+        
+        listeners.append(listener)
+
+        return listener
+    }
+
+    public func removeListener(_ listener: Any) {
+        guard let beaconInfoSummaryListener = listener as? MXBeaconInfoSummaryListener else {
+            return
+        }
+        
+        listeners.removeAll(where: { $0 === beaconInfoSummaryListener })
+    }
+    
+    // MARK: - Private
+    
+    private func addOrUpdateBeaconInfo(_ beaconInfo: MXBeaconInfo, inRoomWithId roomId: String) {
+        
+        guard let eventId = beaconInfo.originalEvent?.eventId else {
+            return
+        }
+        
+        var beaconInfoSummary: MXBeaconInfoSummary?
+        
+        if let existingBeaconInfoSummary = self.getBeaconInfoSummary(withIdentifier: eventId, inRoomWithId: roomId) {
+
+            // If beacon info is older than existing one, do not take it into account
+            if beaconInfo.timestamp > existingBeaconInfoSummary.beaconInfo.timestamp {
+                existingBeaconInfoSummary.updateWithBeaconInfo(beaconInfo)
+                beaconInfoSummary = existingBeaconInfoSummary
+            }
+        } else {
+            beaconInfoSummary = MXBeaconInfoSummary(beaconInfo: beaconInfo)
+        }
+        
+        if let beaconInfoSummary = beaconInfoSummary {
+            self.beaconInfoSummaryStore.addOrUpdateBeaconInfoSummary(beaconInfoSummary, inRoomWithId: roomId)
+            
+            self.notifyBeaconInfoSummaryListeners(ofRoomWithId: roomId, beaconInfoSummary: beaconInfoSummary)
+        }
+    }
+    
+    private func canAddBeacon(_ beacon: MXBeacon, to beaconInfoSummary: MXBeaconInfoSummary) -> Bool {
+    
+        guard beaconInfoSummary.hasStopped == false, beaconInfoSummary.hasExpired == false,
+        beacon.timestamp < beaconInfoSummary.expiringDate else {
```

Comment:
Should probably have each condition on its own line, or all on the same line here.